service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -268,7 +268,7 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
                 if (endPoint.getRequestParamWithLimit() != null && requestParamWithLimit.equals(arg)) {
                     arg = listElementVarName;
                 }
-                if (parameter.getType().equals("String")) {
+                if (parameter.getType().equals("String") && parameter.getEncode()) {
```

Comment:
The default value of this true, so it doesn't affect anywhere. Only specific parameters where the behaviour is not desired can be now turned off using the flag.